### ISSUE TITLE
Notion of no. possible unique sequences with readers.video

### ISSUE THREAD

**Issue by Michelvl92** [NVIDIA/DALI#3046]
Hi,

I have followed the tutorial from [Video Pipeline Reading Labelled Videos from a Directory](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/sequence_processing/video/video_reader_label_example.html), which I use to train a model in Tensorflow/Keras together with the tf plugin: [nvidia.dali.plugin.tf.DALIDataset](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/plugins/tensorflow_plugin_api.html#nvidia.dali.plugin.tf.DALIDataset) where I pass the dataset to model created with `tf.keras` and use `model.fit` method to train it.

There is a fixed no of iterations defined. For my custom dataset, I want to know how many possible unique sequences are created in total, such that during training it would it be possible to use all the unique sequences for one epoch?

If this is possible would you like to provide an example?

**Comment by Michelvl92** [comment]
Hi,

I have followed the tutorial from: https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/sequence_processing/video/video_reader_label_example.html

There is a fixed no of iterations defined. For my custom dataset I want to know how many possible sequences are created in total, such that during training with tensorflow/Keras, and would it be possible to use all the sequences for one epoch?

If this is possible would you like to provide an example?

**Comment by JanuszL** [comment]
Hi @Michelvl92,

You can use [`reader_meta`](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/pipeline.html#nvidia.dali.Pipeline.reader_meta) pipeline method to learn how many samples would DALI return. If you divide it by the batch size then you should get the number of epochs. If the epoch size is not divisible DALI would just wrap the data set and read from the next epoch, or you can use the `pad_last_batch` argument to duplicate the last sample.

**Comment by Michelvl92** [comment]
Perfect! That makes sense, wasn't aware of that. 

It's only not clear what you mean with "divide it by the batch size then you should get the number of epochs". I would then expect that changing the `batch_size`, e.g. from 1 --> 2 would stay the same. Or should I interpret that `epoch_size` is independent of the `batch_size`?

It seems that `pad_last_batch=True` doest increase the `epoch_size`? E.g. video with 150 frames, `step=-1`,` stride=1`, `sequence_length=25` sets epoch_size = 6, while `sequence_length=26` sets epoch_size = 5 (`pad_last_batch': 1`) while I would expect to duplicate the last sample, and thus  epoch_size = 6.

**Comment by JanuszL** [comment]
Hi @Michelvl92,

> It's only not clear what you mean with "divide it by the batch size then you should get the number of epochs".

My bad. I was thinking about the number of iterations in the epoch, and the data set size divisible by the number of shards.

And the metadata you can get is:
![image](https://user-images.githubusercontent.com/39967756/121941800-00d57900-cd50-11eb-89ac-2b46328b70fd.png)

So to make sure that shards are equal, use `pad_last_batch=True`. And then `ceils(epoch_size_padded / num_shards / batch_size)` should be the number of iteration in each epoch.

Padding last bach is orthogonal to `sequence_length`, it pads only valid samples (full sequences in this case), not missing frames inside the sequence. There is a community, WIP PR that should address this. If you like you can continue the work as the original author doesn't have time to finish it.
